kong/plugins/prometheus/exporter.lua

```diff
@@ -324,11 +324,11 @@ local function metric_data()
   -- memory stats
   local res = kong.node.get_memory_stats()
   for shm_name, value in pairs(res.lua_shared_dicts) do
-    metrics.memory_stats.shms:set(value.allocated_slabs, {shm_name})
+    metrics.memory_stats.shms:set(value.allocated_slabs, { shm_name, ngx.config.subsystem })
   end
   for i = 1, #res.workers_lua_vms do
     metrics.memory_stats.worker_vms:set(res.workers_lua_vms[i].http_allocated_gc,
-                                        {res.workers_lua_vms[i].pid})
+                                        { res.workers_lua_vms[i].pid, ngx.config.subsystem })
```

Comment:
nit: If there is a performance penalty for not caching the subsystem, then please cache it at the module level. If not, please ignore my comment.

Reply:
done.